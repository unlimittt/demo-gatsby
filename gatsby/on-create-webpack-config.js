const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
