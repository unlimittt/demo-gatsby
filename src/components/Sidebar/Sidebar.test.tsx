import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Sidebar from './Sidebar';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

describe('Sidebar', () => {
  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(siteMetadata)
    );
  });

  const props = {
    isIndex: true
  };

  it('renders correctly', () => {
    const wrapper = shallow(<Sidebar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
