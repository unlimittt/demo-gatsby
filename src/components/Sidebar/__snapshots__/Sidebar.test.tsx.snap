// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="sidebar__inner"
  >
    <Author
      author={
        Object {
          "bio": "Test bio",
          "contacts": Object {
            "email": "#",
            "github": "#",
            "rss": "#",
            "telegram": "#",
            "twitter": "#",
            "vkontakte": "#",
          },
          "name": "Test name",
          "photo": "/test.jpg",
        }
      }
      isIndex={true}
    />
    <Menu
      menu={
        Array [
          Object {
            "label": "Test label 1",
            "path": "/test/1/",
          },
          Object {
            "label": "Test label 2",
            "path": "/test/2/",
          },
          Object {
            "label": "Test label 3",
            "path": "/test/3/",
          },
        ]
      }
    />
    <Contacts
      contacts={
        Object {
          "email": "#",
          "github": "#",
          "rss": "#",
          "telegram": "#",
          "twitter": "#",
          "vkontakte": "#",
        }
      }
    />
    <Copyright
      copyright="Test copyright"
    />
  </div>
</div>
`;
